### PR TITLE
Add TLAPS and TLA+ Toolbox

### PR DESCRIPTION
**Pull request by badmutex**
###### Motivation for this change
This attempts to address #18640.

Testing:
- `tlaplusToolbox`: I've been using this without unexpected issues (pdf generation needs `pdflatex` in the `PATH`). 
- `tlaps`: Both `tlapm` and `zenon` run and produce their usage information.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/science/logic/tlaplus/tla-toolbox.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Type=Application
+Exec=tla-toolbox
+Icon=tla-toolbox
+Comment=IDE for TLA+
+Terminal=false
+Name=TLA Toolbox
+Categories=Application;Development;
+StartupWMClass=TLA+ Toolbox
```
Comment (Mic92) [inline pkgs/applications/science/logic/tlaplus/tla-toolbox.desktop]: For desktop items we usually use this: 

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/browsers/chromium/default.nix#L37

Comment (badi) [reply pkgs/applications/science/logic/tlaplus/tla-toolbox.desktop]: ooh that's useful. I'll do that. Thanks for the pointer!

File: pkgs/applications/science/logic/tlaplus/tlaps.nix [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{ lib
+, fetchurl
+, makeWrapper
+, stdenv
+, ocaml, gawk, isabelle, cvc3, perl, wget, which
+, ...
```
Comment (Mic92) [inline pkgs/applications/science/logic/tlaplus/tlaps.nix]: `...` should be removed.

File: pkgs/applications/science/logic/tlaplus/toolbox.nix [hunk 3]
```diff
@@ -0,0 +1,80 @@
+{ lib, fetchzip, makeWrapper, makeDesktopItem, stdenv, jre, swt, gtk, libXtst, glib
+, ... }:
```
Comment (Mic92) [inline pkgs/applications/science/logic/tlaplus/toolbox.nix]: also here.

File: pkgs/applications/science/logic/tlaplus/toolbox.nix [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    patchelf \
+      --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) \
+      "$out/toolbox/toolbox"
+
+    echo "cd $(echo $out/toolbox); ./toolbox -data ~/.tla-toolbox \"$@\"" \
+      > "$out/bin/tla-toolbox"
+
+    chmod +x $out/bin/tla-toolbox
+
+    wrapProgram "$out/bin/tla-toolbox" \
```
Comment (Mic92) [inline pkgs/applications/science/logic/tlaplus/toolbox.nix]: if you use `makeWrapper`, you don't need the `echo` above. It also has a `--run` parameter.

**Pull request merged**